backend/src/services/auto_trader.py

```
@@ -120,151 +103,103 @@ async def _scan_and_buy(self):
-                current_token_data = await self.data_fetcher_service.get_token_by_address(token_address)
-                if not current_token_data:
-                    logger.warning(f"AutoTrader: Owned token {details.get('symbol', token_address)} not found in data fetcher. Skipping sell monitoring.")
+                # Real balance check
+                current_balance = await self.wallet_service.get_token_balance(token_address)
+                if current_balance <= 0:
+                    tokens_to_remove.append(token_address)
                     continue
 
+                details['current_amount_tokens'] = current_balance
+                current_token_data = await self.data_fetcher_service.get_token_by_address(token_address)
+                if not current_token_data: continue
```

Ruff (0.15.6)

[error] Multiple statements on one line (colon)

(E701)